_posts/2024-07-03-oidc-cient-request.adoc

```diff
@@ -0,0 +1,447 @@
+
+=== OpenShift CLI
+To make sure your Keycloak server has been provisioned using the OpenShift CLI, run:
+```
+oc get pods
+```
+
+After a little while, check for a message similar to the following message that indicates the pod is ready:
+```
+NAME                READY     STATUS      RESTARTS   AGE
+keycloak-1-deploy   0/1       Completed   0          1h
+keycloak-1-l9kdx    1/1       Running     0          1h
+```
+
+Once the Keycloak server has been provisioned, use the following command to find the URL for your Keycloak instance’s Admin Console:
+```
+KEYCLOAK_URL=https://$(oc get route keycloak --template='{{ .spec.host }}') &&
+echo "" &&
+echo "Keycloak Admin Console:   $KEYCLOAK_URL/admin" &&
+echo ""
+```
+
+=== OpenShift Web Console
+To make sure your Keycloak server has been provisioned using the OpenShift web console, navigate to the *Topology* view in the *Developer* perspective. You can click on your *keycloak* app to check its status. Once it is running, you can click on *Open URL* and you will be taken to Keycloak’s *Administration Console* login page.
+
+== Setting up Keycloak OpenID Provider
+
+. Log into the Keycloak Admin Console using the admin username and password you specified earlier. 
+. Next, create a realm called *myrealm* and register a client called *myclient* as follows: 
+*General settings*:
+** *Client type* (or *Client Protocol*, depending on your keycloak version): *OpenID Connect*
+** *Client ID*: *myclient*
+* *Capability config*:
+** *Client authentication* : *On*
+** *Authentication flow*: *Standard flow, Direct access grants*
+* *Login Settings*: For the *Valid Redirect URIs*, enter a * for now. We will come back to edit it later.
+
+. Once you hit *Save*, you should see a new tab called *Keys* appear.
+
+. Navigate to that tab and click on the *Import* button located at the bottom of the screen. 
+* Under *Archive format*, choose *PKCS12* from the dropdown. 
+* Under *Key alias*, enter the alias of the RSA key you just created. 
+* Under *Store password* enter the password of the keystore you created
+and click on the *Browse* button to import the keystore file. Note that when we created the keystore, we specified a key
+password and a keystore password. Here, we are only using the *keystore* password. While we set them both to be the same, they do not have to be.
+* Once you select the keystore file created earlier, click *Import* and you should see a message at the top of the screen indicating that the certificate has been uploaded successfully and you will see it in the text field in the middle of the screen.
+
+. Finally, create a user called *alice* whose *First name* is *Alice* and *Last name* is *Smith*. You can find more details about creating and managing KeyCloak users https://www.keycloak.org/docs/latest/server_admin/#proc-creating-user_server_administration_guide[here].
```

Comment:
These directions will help the user create a new user more quickly.
-- create a new user for realm, myrealm
Select the drop down menu in the upper left corner
Select the "User" menu item.
On the "Users" screen click the "Create new user" button
In the "Username" box type the user alias, "alice"
In the "First name" box type "Alice"
In the "Last name" box type "Smith"
Click the "Create" button

-- set Alice's account password
On alice's "User details" page, click on the "Credentials" tab.
Click the "Set password" button.
In the "Set password" dialog, set a password of your choosing for alice.
Make note of this password.  It will be required later
Toggle the `Temporary` switch to off, so you are not prompted to update 
the password after the first login.
Click the "Save" button